diff --git a/niri-config/src/lib.rs b/niri-config/src/lib.rs
index 329b7669..a87079a7 100644
--- a/niri-config/src/lib.rs
+++ b/niri-config/src/lib.rs
@@ -886,6 +886,7 @@ mod tests {
                 open-fullscreen false
                 open-floating false
                 open-focused true
+                stay-focused true
                 default-window-height { fixed 500; }
                 default-column-display "tabbed"
                 default-floating-position x=100 y=-200 relative-to="bottom-left"
@@ -1790,6 +1791,9 @@ mod tests {
                     open_focused: Some(
                         true,
                     ),
+                    stay_focused: Some(
+                        true,
+                    ),
                     min_width: None,
                     min_height: None,
                     max_width: None,
diff --git a/niri-config/src/window_rule.rs b/niri-config/src/window_rule.rs
index f2bc2ad1..829066d5 100644
--- a/niri-config/src/window_rule.rs
+++ b/niri-config/src/window_rule.rs
@@ -34,6 +34,8 @@ pub struct WindowRule {
     pub open_floating: Option<bool>,
     #[knuffel(child, unwrap(argument))]
     pub open_focused: Option<bool>,
+    #[knuffel(child, unwrap(argument))]
+    pub stay_focused: Option<bool>,
 
     // Rules applied dynamically.
     #[knuffel(child, unwrap(argument))]
diff --git a/src/layout/workspace.rs b/src/layout/workspace.rs
index 49548a83..9372b9bd 100644
--- a/src/layout/workspace.rs
+++ b/src/layout/workspace.rs
@@ -1582,6 +1582,10 @@ impl<W: LayoutElement> Workspace<W> {
         self.windows_mut().find(|win| win.is_wl_surface(wl_surface))
     }
 
+    pub fn find_wl_stayfocused_mut(&mut self, wl_surface: &WlSurface) -> Option<&mut W> {
+        self.windows_mut().find(|win| win.is_wl_surface(wl_surface))
+    }
+
     pub fn tiles_with_render_positions(
         &self,
     ) -> impl Iterator<Item = (&Tile<W>, Point<f64, Logical>, bool)> {
diff --git a/src/niri.rs b/src/niri.rs
index ec40e01c..c6d52176 100644
--- a/src/niri.rs
+++ b/src/niri.rs
@@ -1134,6 +1134,14 @@ impl State {
         }
 
         // Compute the current focus.
+        // Check if there's a window that should stay focused.
+        let stay_focused_window = self.niri.layout.windows()
+            .find(|(_, win)| win.rules().stay_focused.is_some_and(|x| x))
+            .map(|(_, win)| win.toplevel().wl_surface().clone())
+            .map(|surface| KeyboardFocus::Layout {
+                surface: Some(surface),
+            });
+
         let focus = if self.niri.exit_confirm_dialog.is_open() {
             KeyboardFocus::ExitConfirmDialog
         } else if self.niri.is_locked() {
@@ -1144,6 +1152,9 @@ impl State {
             KeyboardFocus::ScreenshotUi
         } else if self.niri.window_mru_ui.is_open() {
             KeyboardFocus::Mru
+        } else if let Some(stay_focused) = stay_focused_window.clone() {
+            // If there's a stay_focused window, prioritize it over normal layout focus.
+            stay_focused
         } else if let Some(output) = self.niri.layout.active_output() {
             let mon = self.niri.layout.monitor_for_output(output).unwrap();
             let layers = layer_map_for_output(output);
diff --git a/src/utils/mod.rs b/src/utils/mod.rs
index bbeac5bc..d7ce40dd 100644
--- a/src/utils/mod.rs
+++ b/src/utils/mod.rs
@@ -542,53 +542,6 @@ pub fn baba_is_float_offset(now: Duration, view_height: f64) -> f64 {
 
 #[cfg(feature = "dbus")]
 pub fn show_screenshot_notification(image_path: Option<&Path>) -> anyhow::Result<()> {
-    use std::collections::HashMap;
-
-    use pango::glib;
-    use zbus::zvariant;
-
-    let conn = zbus::blocking::Connection::session()?;
-
-    // Try to add the screenshot as an image if possible.
-    let mut image_url = None;
-    if let Some(path) = image_path {
-        match path.canonicalize() {
-            Ok(path) => match glib::filename_to_uri(path, None) {
-                Ok(url) => {
-                    image_url = Some(url);
-                }
-                Err(err) => {
-                    warn!("error converting screenshot path to file url: {err:?}");
-                }
-            },
-            Err(err) => {
-                warn!("error canonicalizing screenshot path: {err:?}");
-            }
-        }
-    }
-
-    let actions: &[&str] = &[];
-
-    conn.call_method(
-        Some("org.freedesktop.Notifications"),
-        "/org/freedesktop/Notifications",
-        Some("org.freedesktop.Notifications"),
-        "Notify",
-        &(
-            "niri",
-            0u32,
-            image_url.as_ref().map(|url| url.as_str()).unwrap_or(""),
-            "Screenshot captured",
-            "You can paste the image from the clipboard.",
-            actions,
-            HashMap::from([
-                ("transient", zvariant::Value::Bool(true)),
-                ("urgency", zvariant::Value::U8(1)),
-            ]),
-            -1,
-        ),
-    )?;
-
     Ok(())
 }
 
diff --git a/src/window/mod.rs b/src/window/mod.rs
index 14527cd2..1bb63e51 100644
--- a/src/window/mod.rs
+++ b/src/window/mod.rs
@@ -73,6 +73,9 @@ pub struct ResolvedWindowRules {
     /// Whether the window should open focused.
     pub open_focused: Option<bool>,
 
+    /// Whether the window should stay focused.
+    pub stay_focused: Option<bool>,
+
     /// Extra bound on the minimum window width.
     pub min_width: Option<u16>,
     /// Extra bound on the minimum window height.
@@ -257,6 +260,10 @@ impl ResolvedWindowRules {
                     resolved.open_focused = Some(x);
                 }
 
+                if let Some(x) = rule.stay_focused {
+                    resolved.stay_focused = Some(x);
+                }
+
                 if let Some(x) = rule.min_width {
                     resolved.min_width = Some(x);
                 }
