diff --git a/osc.c b/osc.c
index 4dc47172..da14143c 100644
--- a/osc.c
+++ b/osc.c
@@ -761,7 +761,7 @@ kitty_notification(struct terminal *term, char *string)
 
                 const char *terminator = term->vt.osc.bel ? "\a" : "\033\\";
 
-                char reply[128];
+                char reply[512];
                 size_t n = xsnprintf(
                     reply, sizeof(reply),
                     "\033]99;i=%s:p=?;p=%s:a=%s:o=%s:u=%s:c=1:w=1:s=system,silent,error,warn,warning,info,question%s",
