diff -urNa pidgin-sipe-1.18.4/src/purple/purple-buddy.c pidgin-sipe-1.18.4_mod/src/purple/purple-buddy.c
--- pidgin-sipe-1.18.4/src/purple/purple-buddy.c	2014-10-18 22:10:05.000000000 +0800
+++ pidgin-sipe-1.18.4_mod/src/purple/purple-buddy.c	2014-11-21 16:31:01.283642427 +0800
@@ -138,7 +138,15 @@
 gchar* sipe_backend_buddy_get_alias(SIPE_UNUSED_PARAMETER struct sipe_core_public *sipe_public,
 				    const sipe_backend_buddy who)
 {
-	return g_strdup(purple_buddy_get_alias(who));
+	gchar* alias = purple_buddy_get_alias(who);
+	if (!alias || alias[0] == 0) {
+		alias = purple_buddy_get_name(who);
+	}
+	if (alias && alias[0]) {
+		return g_strdup(alias);
+	}
+
+	return g_strdup("Unamed.");
 }
 
 gchar* sipe_backend_buddy_get_server_alias(SIPE_UNUSED_PARAMETER struct sipe_core_public *sipe_public,
diff -urNa pidgin-sipe-1.18.4/src/telepathy/telepathy-buddy.c pidgin-sipe-1.18.4_mod/src/telepathy/telepathy-buddy.c
--- pidgin-sipe-1.18.4/src/telepathy/telepathy-buddy.c	2012-12-02 01:40:12.000000000 +0800
+++ pidgin-sipe-1.18.4_mod/src/telepathy/telepathy-buddy.c	2014-11-21 16:30:34.810642856 +0800
@@ -650,7 +650,15 @@
 gchar *sipe_backend_buddy_get_alias(SIPE_UNUSED_PARAMETER struct sipe_core_public *sipe_public,
 				    const sipe_backend_buddy who)
 {
-	return(g_strdup(((struct telepathy_buddy_entry *) who)->buddy->info[SIPE_BUDDY_INFO_DISPLAY_NAME]));
+	gchar* alias = ((struct telepathy_buddy_entry *) who)->buddy->info[SIPE_BUDDY_INFO_DISPLAY_NAME];
+	if (alias && alias[0] != 0) {
+		return(g_strdup(alias));
+	}
+	alias = sipe_backend_buddy_get_name(sipe_public, who);
+	if (alias && alias[0] != 0) {
+		return(g_strdup(alias));
+	}
+	return g_strdup("Unamed.");
 }
 
 gchar *sipe_backend_buddy_get_server_alias(struct sipe_core_public *sipe_public,
