From ae98f974e0a111722a3f05703bf8979156b88496 Mon Sep 17 00:00:00 2001
From: busyo <52122634+busyoGG@users.noreply.github.com>
Date: Thu, 27 Nov 2025 23:48:34 +0800
Subject: [PATCH] fix: popup position (#281)

---
 src/server/mod.rs | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/server/mod.rs b/src/server/mod.rs
index 4bead43..f972e00 100644
--- a/src/server/mod.rs
+++ b/src/server/mod.rs
@@ -944,8 +944,12 @@ impl<S: X11Selection + 'static> InnerServerState<S> {
         };
         if dims == win.attrs.dims {
             return;
+        } else if win.attrs.is_popup {
+            win.attrs.dims = dims;
         }
+
         debug!("Reconfiguring {:?} {:?}", event.window(), dims);
+
         if !win.mapped {
             win.attrs.dims = dims;
             return;
-- 
2.51.2

