diff -urNa xkbcomp-1.4.7/keycodes.c xkbcomp-1.4.7-modified/keycodes.c
--- xkbcomp-1.4.7/keycodes.c	2024-02-04 06:31:40.000000000 +0800
+++ xkbcomp-1.4.7-modified/keycodes.c	2024-08-21 17:18:17.069428803 +0800
@@ -656,8 +656,8 @@
     }
     if ((tmp.ival > XkbMaxLegalKeyCode))
     {
-        WARN("Unsupported maximum keycode %d, clipping.\n", tmp.ival);
-        ACTION("X11 cannot support keycodes above 255.\n");
+        /* WARN("Unsupported maximum keycode %d, clipping.\n", tmp.ival); */
+        /* ACTION("X11 cannot support keycodes above 255.\n"); */
         info->explicitMax = XkbMaxLegalKeyCode;
         info->effectiveMax = XkbMaxLegalKeyCode;
         return 1;
diff -urNa xkbcomp-1.4.7/symbols.c xkbcomp-1.4.7-modified/symbols.c
--- xkbcomp-1.4.7/symbols.c	2024-02-04 06:31:40.000000000 +0800
+++ xkbcomp-1.4.7-modified/symbols.c	2024-08-21 17:45:25.228475319 +0800
@@ -946,11 +946,11 @@
     key->symsDefined |= (1U << ndx);
     for (int i = 0; i < nSyms; i++) {
         if (!LookupKeysym(value->value.list.syms[i], &key->syms[ndx][i])) {
-            if (warningLevel > 0)
-            {
-                WARN("Could not resolve keysym %s\n",
-                      value->value.list.syms[i]);
-            }
+            /* if (warningLevel > 0) */
+            /* { */
+            /*     WARN("Could not resolve keysym %s\n", */
+            /*           value->value.list.syms[i]); */
+            /* } */
             key->syms[ndx][i] = NoSymbol;
         }
     }
